src/util/temp-dir.js

```diff
@@ -63,7 +66,9 @@ export class TempDir {
    * been created.
    */
   create(): Promise<TempDir> {
-    const createTempDir = promisify(tmp.dir, tmp, {multiArgs: true});
+    //const createTempDir = promisify(tmp.dir, tmp, {multiArgs: true});
+    const createTempDir = promisify(tmp.dir);
```

Comment:
This whole spread-out method (separate utility file, overriding `promisify.custom` of an *external* library at the top of the file and additional `promisify` invocation here before finally calling the method looks a bit ugly.

Since this is the only user of `util/promisify`, how about just putting the non-generic logic here? Like this:

```javascript
return new Promise((resolve, reject) => {
  tmp.dir({
    ...
  }, (err, tmpPath, removeTempDir) => {
    if (err) {
      reject(err);
      return;
    }
    this._path = tmpPath;
    this._removeTempDir = removeTempDir;
    log.debug(`Created temporary directory: ${this.path()}`);
    resolve(this);
  });
});
````

Reply:
Well, we could move the definition of `createTempFile` near its overriding with `promisify.custom`,
that way we would keep most of it in one place.

I just added c275812 to apply the change described above, let me know how that looks to you.

Reply:
That's looking better, and I was about to suggest that before I pondered about just using promises directly. Using promises directly doesn't look less readable (if any, it looks more readable to me).

If you like keeping `util/promisify.js`, then I would suggest one more change:

```
.then((args) => {
        const [tmpPath, removeTempDir] = args;
```

to

```
.then(([tmpPath, removeTempDir]) => {
```